sstables/sstables.cc

```diff
@@ -2729,7 +2728,7 @@ remove_by_toc_name(sstring sstable_toc_name) {
             sstlog.debug("Forgiving ENOENT when deleting file {}", fname);
         }
     });
-    co_await fsync_directory(sstable_write_error_handler, dir);
+    co_await fsync_directory(sstable_write_error_handler, parent_path(new_toc_name));
```

Comment:
I still I don't understand why in the first case of the if(), you want to run exactly the same fsync_directory() call twice - once at the end of the if(), and once here right after ending the if! What am I missing?

Reply:
If we crash, we want to make sure the TOC file was renamed to the temporary TOC name, first, before deleting any other components, since this is the commit point.
We want to avoid ending up with the TOC name and missing components.

Reply:
It's one fsync after renaming TOC to TMP TOC, and one fsync after removing all components *and* before removing TMP TOC.

Reply:
@nyh See commit for a simplified form of this function when there was no TMP TOC: 851248e263256069768e96f32494a8fe999bc203